frontend/src/components/custom-tools/export-tool/ExportTool.jsx

```diff
@@ -0,0 +1,228 @@
+import {
+  DeleteOutlined,
+  QuestionCircleOutlined,
+  UserOutlined,
+} from "@ant-design/icons";
+import {
+  Avatar,
+  List,
+  Modal,
+  Popconfirm,
+  Radio,
+  Select,
+  Typography,
+} from "antd";
+import PropTypes from "prop-types";
+import { useEffect, useState } from "react";
+import "./ExportTool.css";
+
+import { SpinnerLoader } from "../../widgets/spinner-loader/SpinnerLoader";
+
+const SHARE_ALL = "share_all";
+const SHARE_CUSTOM = "share_custom";
+
+function ExportTool({
+  open,
+  setOpen,
+  toolDetails,
+
+  loading,
+  allUsers,
+  onApply,
+}) {
+  const [selectedUsers, setSelectedUsers] = useState([]);
+  const [filteredUserList, setFilteredUserList] = useState([]);
+  const [sharingOption, setSharingOption] = useState(SHARE_ALL);
+
+  useEffect(() => {
+    if (toolDetails?.shared_users) {
```

Comment:
@johnyrahul NIT: Can we follow the below approach to avoid nesting level? 

```
if (!toolDetails?.shared_users) {
    return;
}

// Main Logic
```